syntax = "proto3";
package CompB;

// ******************************************************************************************** //
message AMessage
{
    message Astruct
    {
        int32 A_01 = 1;
        float A_02 = 2;
    }
    repeated Astruct A = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** //
message EMessage
{
    bool E = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** //
message FMessage
{
    double F = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** //
message GMessage
{
    double G = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************* //
message HMessage
{
    repeated float H = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** //
message IMessage
{
    message Istruct
    {
        message Istructstruct
        {
        int32 I_01_01 = 1;
        float I_01_02 = 2;
        }
        repeated Istructstruct I = 1;
    }
    repeated Istruct Ist = 1;
}
// ******************************************************************************************** //

// ******************************************************************************************** // 
message CompBInMessage
{
    AMessage A = 1;
    FMessage F = 2;
    HMessage H = 3;
}
// ******************************************************************************************** // 

// ******************************************************************************************** // 
message CompBOutMessage
{
    EMessage E = 1;
    GMessage G = 2;
    IMessage I = 3;
}
// ******************************************************************************************** // 
